maro/cli/inspector/cim_dashboard.py

```diff
@@ -0,0 +1,546 @@
+    # Generate data.
+    data = helper.read_detail_csv(os.path.join(source_path, GlobalFilePaths.ports_sum)).iloc[down_pooling_range]
+    data["remaining_space"] = list(
+        map(
+            lambda x, y, z: x - y - z,
+            data["capacity"],
+            data["full"],
+            data["empty"]
+            )
+        )
+    # Get formula and selected data.
+    filtered_data = helper.get_filtered_formula_and_data(GlobalScenarios.CIM, data, option_candidates)
+    data = filtered_data["data"]
+    filtered_option = filtered_data["item_option"]
+    _generate_inter_view_panel(data[filtered_option], down_pooling_range)
+
+
+def render_intra_view(source_path: str, epoch_num: int, prefix: str):
+    """Show CIM detail plot.
+
+    Args:
+        source_path (str): The root path of the dumped snapshots data for the corresponding experiment.
+        epoch_num (int) : Total number of epoches,
+                        i.e. the total number of data folders since there is a folder per epoch.
+        prefix (str):  Prefix of data folders.
+    """
+    option_epoch = st.sidebar.select_slider(
+        label="Choose an Epoch:",
+        options=list(range(0, epoch_num))
+    )
+    target_path = os.path.join(source_path, f"{prefix}{option_epoch}")
+    # Get data of selected epoch.
+    data_ports = helper.read_detail_csv(os.path.join(target_path, "ports.csv"))
+    data_ports["remaining_space"] = list(
+        map(
+            lambda x, y, z: x - y - z,
+            data_ports["capacity"],
+            data_ports["full"],
+            data_ports["empty"]
+            )
+        )
+    # Basic data.
+    ports_num = len(data_ports["name"].unique())
+    ports_index = np.arange(ports_num).tolist()
+    snapshot_num = len(data_ports["frame_index"].unique())
+    snapshots_index = np.arange(snapshot_num).tolist()
+
+    # Items for user to select.
+    option_candidates = CIMItemOption.quick_info + CIMItemOption.booking_info + CIMItemOption.port_info
+    # Name conversion.
+    name_conversion = helper.read_detail_csv(os.path.join(source_path, GlobalFilePaths.name_convert))
+
+    st.sidebar.markdown("***")
+    option_view = st.sidebar.selectbox(
+        label="By ports/snapshot:",
+        options=CIMIntraViewChoice._member_names_
+    )
+
+    if option_view == CIMIntraViewChoice.by_port.name:
+        _render_intra_view_by_ports(
+            data_ports, ports_index, name_conversion,
+            option_candidates, snapshot_num)
+    elif option_view == CIMIntraViewChoice.by_snapshot.name:
+        _render_intra_view_by_snapshot(
+            source_path, option_epoch, data_ports, snapshots_index,
+            name_conversion, option_candidates, ports_num, prefix)
+    else:
+        pass
+
+
+def _generate_inter_view_panel(data: pd.DataFrame, down_pooling_range: list):
+    """Generate summary plot.
+
+        View info within different epochs.
```

Comment:
No indents are needed for the information details. That:
```
"""Generate summary plot.

View info within different epochs.

Args:
```

Reply:
For your reference: https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html